tests/test_expectation_maximization.py

```diff
@@ -90,7 +90,23 @@ def test_grid_SO3_uniform(test_ir, n_particles):
 def test_generate_xy_plane(test_ir, n_pix):
     """Test generation of xy plane."""
     xy_plane = test_ir.generate_xy_plane(n_pix)
-    assert xy_plane.shape == (n_pix**2, 3)
+    assert xy_plane.shape == (3, n_pix**2)
+
+    plane_2 = np.array([[-0.5, 0.5, -0.5, 0.5], [-0.5, -0.5, 0.5, 0.5], [0, 0, 0, 0]])
```

Comment:
Change this to reflect the convention `(-n//2, ... n//2-1)`

Reply:
Otherwise `assert np.allclose(xy_plane, plane_2)` will fail.

Reply:
@geoffwoollard I think this is something worth discussing. How do we rotate given that convention? When you apply a rotation matrix to this convention, it is rotating about an offset axis (namely about the row of index n // 2 + 1 where the values are 0). 

This means that rotating by 180 degrees vs rotating by -180 degrees produces different results (and not just in an orientation sense) as the array is offset by a different amount depending on the direction of rotation.

Reply:
> This means that rotating by 180 degrees vs rotating by -180 degrees produces different results (and not just in an orientation sense) as the array is offset by a different amount depending on the direction of rotation.

Upon further reflection I'm not sure of this issue, that was a mistake of interpretation on my part. However, the offset axis is still something I want your thoughts on :)

Reply:
we rotate before we offset the axis. See https://github.com/geoffwoollard/compSPI/blob/stash_simulate/src/simulate.py#L109

Is that what you mean?

Reply:
For generating slices, you can just closely follow https://github.com/geoffwoollard/compSPI/blob/stash_simulate/src/simulate.py#L109 and everything should work out fine! As you go through it to understand things, write up good tests and notes in the documentation. What axis are we taking the slice through in the arr_3d. What does this imply about the coordinates of `arr_3d`. Are they xyz or zxy, zyx??? How does this correspond to the axis of the rotation (usually Rxx on the 11 element and Rzz on the 33 element), and return from `coords_n_by_d`

Reply:
Not exactly. When we rotate an xy plane matrix by 180 degrees about a principal axis, it should remain the same. In a coding sense, this is true of symmetrical matrices, e.g.:
```
[ -.5  .5 -.5 .5 ]    
[ -.5 -.5  .5 .5 ]
[   0   0   0  0 ]    
```
but in our representation, because of the extra "baggage" on one side of the matrix the 180-degree rotation will transform:
```
[ -1  0 -1 0 ]    [ 0 1 0 1 ]
[ -1 -1  0 0 ] -> [ 0 0 1 1 ] 
[  0  0  0 0 ]    [ 0 0 0 0 ]
```
Because it is being rotated about an axis where the plane doesn't have the same amount of coordinates on either side.

Reply:
The reason this error doesn't appear in your notebook is because it uses the geomstats special orthogonal random rotations without accounting for them only covering one hemisphere. So, this process eventually runs in to the issue Fred outlined where they didn't account for particles being viewable from "underneath."

Reply:
@geoffwoollard I'm having considerable difficulty addressing this issue, I expect it to take me some time. Input would be much appreciated. 

As well, I wouldn't mind hearing the rationale behind n_pix being even. Since we need a central rotation axis, and since we have discrete versions of spherical shells with odd-number diameters in expand_1d_to_3d, it seems to me there are many potential conveniences that come from having odd n_pix values.

Reply:
@geoffwoollard issue solved.